File: host/src/main/java/dk/dtu/domain/rules/effects/StartingTile.java
package dk.dtu.domain.rules.effects;
import dk.dtu.domain.core.Phase;
import dk.dtu.domain.model.Board;
import dk.dtu.domain.model.Robot;
import dk.dtu.domain.model.Tile;
import dk.dtu.domain.rules.api.BoardAPI;

import java.util.EnumSet;

/**
 * @author Patrick Røbel
 */
public record StartingTile(int robotId) implements TileEffect {
    @Override
    public void onPhase(Phase phase, Tile tile, BoardAPI api) {
	// probably no effect here?
    }
    @Override
    public EnumSet<Phase> phases() {
        return EnumSet.noneOf(Phase.class);
    }
}

File: engine/src/effects/startingTile.ts
import { Phase } from "../core/phase.js";
import type { Tile } from "../model/tile.js";
import type { BoardAPI } from "../rules/boardApi.js";
import type { TileEffect } from "./tileEffect.js";

/** Ported from dk.dtu.domain.rules.effects.StartingTile. */
export class StartingTile implements TileEffect {
  constructor(readonly robotId: number) {}

  onPhase(_phase: Phase, _tile: Tile, _api: BoardAPI): void {
    // No phase behaviour; used only for initial placement metadata.
  }

  phases(): Set<Phase> | null {
    return new Set();
  }
}
